dkobo/koboform/views/survey_draft_views.py

```diff
@@ -205,27 +225,80 @@ def import_questions(request):
         output[u'error'] = "No file posted"
     return HttpResponse(json.dumps(output), content_type="application/json", status=response_code)
 
-@login_required
+
+def update_xform(formid, public, description, headers):
+    url = kobocat_integration._kobocat_url(
+            '/api/v1/forms/%s' % formid, internal=True)
+    payload = {
+        'public': public,
+        'description': description,
+    }
+
+    response = requests.patch(url, headers=headers, data=payload)
+    if response.status_code in [200, 201]:
+        return {u'message': 'Successfully updated xform'}
+
+    return {'status_code': response.status_code, 'detail': response.text}
+
+
+def create_tags(formid, tags, headers):
+    url = kobocat_integration._kobocat_url(
+                '/api/v1/forms/%s/labels' % formid, internal=True)
+    payload = {
+        'tags': tags
+    }
+    response = requests.post(url, headers=headers, data=payload)
+
+    if response.status_code in [200, 201]:
+        return {u'message': 'Successfully created tags'}
+
+    return {'status_code': response.status_code, 'detail': response.text}
+
+
 @api_view(['GET', 'POST'])
 def publish_survey_draft(request, pk, format=None):
     if not kobocat_integration._is_enabled():
-        return Response({'error': 'KoBoCat Server not specified'}, status=status.HTTP_503_SERVICE_UNAVAILABLE)
+        return Response({'error': 'KoBoCat Server not specified'},
+                        status=status.HTTP_503_SERVICE_UNAVAILABLE)
+
+    myw_kobo_user_cookie = request.COOKIES.get('myw_kobo_user')
```

Comment:
this is almost repeated, maybe it can be abstracted to the helper used by the above similarly repeated code, `token` is the extra bit